src/compile/data/bin.ts

```diff
@@ -89,6 +90,27 @@ export class BinNode extends DataFlowNode {
     return new BinNode(bins);
   }
 
+  public static makeBinFromTransform(model: Model, t: BinTransform) {
+    const bins: Dict<BinComponent> = {};
+
+    if (t.bin) {
+      const bin: Bin = isBoolean(t.bin) ? {} : t.bin;
+      const key = `${binToString(t.bin)}_${t.field}`;
```

Comment:
(You might wanna extract `normalizeBin` method from `normalize` in `fielddef.ts` and put that in `bin.ts` so that we can reuse them in both places)